core/src/workflows/get_link_result.rs

```diff
@@ -33,42 +26,85 @@ pub async fn get_link_result_workflow<'a>(
-                        |link_entry_result| {
-                            link_entry_result
-                                .map(|link_entry_option| {
-                                    link_entry_option.map(|link_entry| {
-                                        let headers = if link_args.options.headers {
-                                            link_entry.headers
-                                        } else {
-                                            Vec::new()
-                                        };
-                                        Ok(LinksResult {
-                                            address: link.clone(),
-                                            headers,
-                                        })
-                                    })
-                                })
-                                .unwrap_or(None)
-                        },
-                    ),
-                )
-                .unwrap_or(Err(HolochainError::ErrorGeneric(
-                    "Could not get links".to_string(),
-                )))
+            get_latest_entry(
+                context.clone(),
+                link.clone(),
+                link_args.options.timeout.clone(),
+            )
+            .map(|link_entry_result_option| {
+                link_entry_result_option.map(|link_entry| {
+                    let headers = if link_args.options.headers {
+                        link_entry.headers
+                    } else {
+                        Vec::new()
+                    };
+                    Ok(LinksResult {
+                        address: link.clone(),
+                        headers,
+                        crud_status: link_entry.entry_with_meta.crud_status,
+                        crud_link: link_entry.entry_with_meta.maybe_link_update_delete,
+                    })
+                })
+            })
+            .unwrap_or(None)
+            .unwrap_or(Err(HolochainError::ErrorGeneric(
+                "Could not crud information for link".to_string(),
+            )))
         })
         .partition(Result::is_ok);
 
     if errors.is_empty() {
         Ok(GetLinksResult::new(
-            link_results.into_iter().map(|s| s.unwrap()).collect(),
+            link_results
+                .into_iter()
+                .map(|s| s.unwrap())
+                .filter(|link_result| match link_args.options.status_request {
+                    LinksStatusRequestKind::All => true,
+                    LinksStatusRequestKind::Deleted => {
+                        link_result.crud_status == CrudStatus::Deleted
+                    }
+                    LinksStatusRequestKind::Live => link_result.crud_status == CrudStatus::Live,
+                })
+                .collect(),
         ))
     } else {
         Err(HolochainError::ErrorGeneric(
             "Could not get links".to_string(),
         ))
     }
 }
+
+pub fn get_latest_entry(
```

Comment:
What exactly is the purpose of this function and why can't we reuse `get_entry_result_workflow`: https://github.com/holochain/holochain-rust/blob/cbd91ef03bf0d06dcc42bf8d0e794ebc2f8ec723/core/src/workflows/get_entry_result.rs#L55 ?

Reply:
I implemented this because we do it have a status request for entries on the getlinks. I forgot to mention it but linkstatus request seems to he on the link level as a whole and entry seems to be referring to the link targets. In general it is just a recursive and non mutable way of doing whatever get_entry_result does and would prefer if we used this one instead but we would need some refactoring